espresso-support/core/src/main/java/com/novoda/espresso/AccessibilityViewMatchers.java

```diff
@@ -0,0 +1,178 @@
+package com.novoda.espresso;
+
+import android.content.res.Resources;
+import android.support.annotation.StringRes;
+import android.view.View;
+import android.view.accessibility.AccessibilityNodeInfo;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import static org.hamcrest.Matchers.is;
+
+public class AccessibilityViewMatchers {
+
+    public static Matcher<? super View> withUsageHintOnClick(@StringRes final int resourceId) {
+        return new TypeSafeMatcher<View>() {
+
+            private String resourceName = null;
+            private String expectedText = null;
+
+            @Override
+            protected boolean matchesSafely(View view) {
+                if (!view.isClickable()) {
+                    return false;
+                }
+
+                if (null == this.expectedText) {
+                    try {
+                        expectedText = view.getResources().getString(resourceId);
+                        resourceName = view.getResources().getResourceEntryName(resourceId);
+                    } catch (Resources.NotFoundException ignored) {
+                        // view could be from a context unaware of the resource id.
+                    }
+                }
+                if (null != expectedText) {
+                    AccessibilityNodeInfo accessibilityNodeInfo = view.createAccessibilityNodeInfo();
+                    AccessibilityNodeInfo.AccessibilityAction clickAction = findAction(accessibilityNodeInfo, AccessibilityNodeInfo.ACTION_CLICK);
+
+                    return expectedText.equals(clickAction.getLabel());
```

Comment:
confirmed it's not currently possible pre-21, gonna add a feature request to AOSP